### DISCUSSION TITLE
SWF File Error (Off Topic)

### DISCUSSION THREAD

**Discussion by SomeCodingCoolGuy** [ruffle-rs/ruffle#3911] [General] [answered]
Today I was going to download a game's SWF file to run in the Ruffle desktop extension/app, by my computer said it would harm my computer.

Operating system: Windows 10 OS Version 1909 (Build 18363.1316)
Browser: Microsoft Edge Build 89.0.774.68 (Official build, 64-bit)
User agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.114 Safari/537.36 Edg/89.0.774.68
Ruffle version/download: See https://github.com/ruffle-rs/ruffle/releases/download/nightly-2021-04-06/ruffle_nightly_2021_04_06_windows.zip
Ruffle extension version:  See https://github.com/ruffle-rs/ruffle/releases/download/nightly-2021-04-06/ruffle_nightly_2021_04_06_extension.zip

**Comment by danielhjacobs** [comment, marked as answer]
It's not that it will harm your computer. It's that it **can** harm your computer. The reason browsers discontinued support for Flash is because malicious code can be included in Flash files, and it is very difficult to actually determine whether code in an swf is malicious without thoroughly examining the source code behind it. That said, Ruffle interprets flash files in a way which is secure. If you download the swf file, click "Keep" when the warning about the potential for the file to harm your computer appears, and then open the file with Ruffle **without opening it in Flash**, it should not be able to run malicious code. If you trust the source you download it from, you can assume it wouldn't have run malicious code anyway, but if you're unsure, this should be safe.
